packages/video_player/video_player_avfoundation/darwin/video_player_avfoundation/Sources/video_player_avfoundation/FVPVideoPlayerPlugin.m

```diff
@@ -21,26 +22,29 @@ @interface FVPFrameUpdater : NSObject
 @property(nonatomic, weak, readonly) NSObject<FlutterTextureRegistry> *registry;
 // The output that this updater is managing.
 @property(nonatomic, weak) AVPlayerItemVideoOutput *videoOutput;
-// The last time that has been validated as avaliable according to hasNewPixelBufferForItemTime:.
-@property(nonatomic, assign) CMTime lastKnownAvailableTime;
+// The display link that drives frameUpdater.
+@property(nonatomic) FVPDisplayLink *displayLink;
+// The time interval between screen refresh updates.
+@property(nonatomic) _Atomic CFTimeInterval duration;
```

Comment:
I think it is not documented whether atomic property with primitive type can be lock free so it looked like overkill. Although there are some hints that it probably can be. So probably there is no problem to change it to atomic property.

Reply:
I deem std atomics as readable enough and there is this nice choice of memory ordering but probably yes atomic property would make for simpler code even with that potential little overhead (it is unknown to me what ordering it uses in case it is lock free).

Reply:
> I deem std atomics as readable enough

You are not the only person who will be reading and maintaining this code. Obj-C developers are overwhelmingly going to be more familiar with `atomic` than `_Atomic` (in fact, I don't think I've ever seen `_Atomic` in Obj-C code in literally decades of Obj-C development).

Unless you have benchmark data showing that `atomic` is an issue in practice, please use it.

Reply:
> You are not the only person who will be reading and maintaining this code.

I did not want to imply that. It is part of a sentence which concludes that atomic property will probably have better tradeoff regarding simplicity (readability) vs performance (meaning better choice).